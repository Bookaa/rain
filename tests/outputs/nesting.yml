!program;1
stmts:
- !assn;2
  export: false
  let: true
  lhs: !name;1
    value: lname
  rhs: !str;1
    value: module local
- !assn;2
  export: true
  let: false
  lhs: !name;1
    value: ename
  rhs: !str;1
    value: module export
- !assn;2
  export: false
  let: true
  lhs: !name;1
    value: loc
  rhs: !table;1
    parent: null
- !assn;2
  export: false
  let: false
  lhs: !index;1
    lhs: !name;1
      value: loc
    rhs: !str;1
      value: name
  rhs: !str;1
    value: local
- !assn;2
  export: false
  let: false
  lhs: !index;1
    lhs: !name;1
      value: loc
    rhs: !str;1
      value: nest
  rhs: !table;1
    parent: null
- !assn;2
  export: false
  let: false
  lhs: !index;1
    lhs: !index;1
      lhs: !name;1
        value: loc
      rhs: !str;1
        value: nest
    rhs: !str;1
      value: name
  rhs: !str;1
    value: local nest
- !assn;2
  export: true
  let: false
  lhs: !name;1
    value: exp
  rhs: !table;1
    parent: null
- !assn;2
  export: false
  let: false
  lhs: !index;1
    lhs: !name;1
      value: exp
    rhs: !str;1
      value: name
  rhs: !str;1
    value: exported
- !assn;2
  export: false
  let: false
  lhs: !index;1
    lhs: !name;1
      value: exp
    rhs: !str;1
      value: nest
  rhs: !table;1
    parent: null
- !assn;2
  export: false
  let: false
  lhs: !index;1
    lhs: !index;1
      lhs: !name;1
        value: exp
      rhs: !str;1
        value: nest
    rhs: !str;1
      value: name
  rhs: !str;1
    value: exported nest
- !assn;2
  export: false
  let: true
  lhs: !name;1
    value: main
  rhs: !func;1
    body: !block;1
      stmts:
      - !call;1
        args:
        - !name;1
          value: lname
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !name;1
          value: ename
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !index;1
          lhs: !name;1
            value: loc
          rhs: !str;1
            value: name
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !index;1
          lhs: !index;1
            lhs: !name;1
              value: loc
            rhs: !str;1
              value: nest
          rhs: !str;1
            value: name
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !index;1
          lhs: !name;1
            value: exp
          rhs: !str;1
            value: name
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !index;1
          lhs: !index;1
            lhs: !name;1
              value: exp
            rhs: !str;1
              value: nest
          rhs: !str;1
            value: name
        catch: false
        func: !name;1
          value: print
    params: []
