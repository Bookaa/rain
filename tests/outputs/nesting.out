module local
module export
local
local nest
exported
exported nest
